swap/src/protocol/alice/spot_price.rs

```diff
@@ -0,0 +1,201 @@
+        max_buy: bitcoin::Amount,
+        latest_rate: LR,
+        resume_only: bool,
+    ) -> Self {
+        Self {
+            behaviour: spot_price::Behaviour::new(
+                CborCodec::default(),
+                vec![(SpotPriceProtocol, ProtocolSupport::Inbound)],
+                RequestResponseConfig::default(),
+            ),
+            events: Default::default(),
+            balance,
+            lock_fee,
+            max_buy,
+            latest_rate,
+            resume_only,
+        }
+    }
+
+    pub fn update_balance(&mut self, balance: monero::Amount) {
+        self.balance = balance;
+    }
+
+    fn handle_error_response(
+        &mut self,
+        peer: PeerId,
+        channel: ResponseChannel<spot_price::Response>,
+        error: spot_price::Error,
+    ) {
+        if self
+            .behaviour
+            .send_response(channel, spot_price::Response::Error(error))
+            .is_err()
+        {
+            tracing::debug!(%peer, "Unable to send error response for spot price request");
+        }
+    }
+
+    fn poll<BIE>(
+        &mut self,
+        _cx: &mut Context<'_>,
+        _params: &mut impl PollParameters,
+    ) -> Poll<NetworkBehaviourAction<BIE, OutEvent>> {
+        if let Some(event) = self.events.pop_front() {
+            return Poll::Ready(NetworkBehaviourAction::GenerateEvent(event));
+        }
+
+        // We trust in libp2p to poll us.
+        Poll::Pending
+    }
+}
+
+impl<LR> NetworkBehaviourEventProcess<spot_price::OutEvent> for Behaviour<LR>
+where
+    LR: LatestRate + Send + 'static,
+{
+    fn inject_event(&mut self, event: spot_price::OutEvent) {
+        let (peer, message) = match event {
+            RequestResponseEvent::Message { peer, message } => (peer, message),
+            RequestResponseEvent::OutboundFailure { peer, error, .. } => {
+                tracing::error!(%peer, "Failure sending spot price response: {:#}", error);
+                return;
+            }
+            RequestResponseEvent::InboundFailure { peer, error, .. } => {
+                tracing::warn!(%peer, "Inbound failure when handling spot price request: {:#}", error);
+                return;
+            }
+            RequestResponseEvent::ResponseSent { peer, .. } => {
+                tracing::debug!(%peer, "Spot price response sent");
+                return;
+            }
+        };
+
+        let (request, channel) = match message {
+            RequestResponseMessage::Request {
+                request, channel, ..
+            } => (request, channel),
+            RequestResponseMessage::Response { .. } => {
+                tracing::error!("Unexpected message");
+                return;
+            }
+        };
+
+        if self.resume_only {
+            tracing::warn!(%peer, "Ignoring spot price request from {} because ASB is running in maintenance mode", peer);
+            self.handle_error_response(peer, channel, spot_price::Error::MaintenanceMode);
```

Comment:
Half the codebase calls this `resume-only`, the other half "MaintainanceMode". Find a single name for it? :)

Reply:
There is actually some reasoning behind it - from the CLI's perspective the ASB is in maintenance mode that's what one cannot trade with it, but from the ASB provider point of view it is `resume-only` mode - other things could lead to maintenance mode as well. 

But that seems to be confusing, so maybe I just call it resume-only then because that is what it is at the moment...